### ISSUE TITLE
Promise usage

### ISSUE THREAD

**Issue by benjamingr** [mikeal/r2#10]
Hi, thanks for the nice library! 

In the code you create a `resolvable()`  only to resolve it with an already existing promise which [is not something you typically have to do](https://stackoverflow.com/questions/23803743/what-is-the-explicit-promise-construction-antipattern-and-how-do-i-avoid-it?noredirect=1#comment78892479_23803743). 

You can use `fetch` as `this.response`, the `setTimeout` also isn't necessary as promises take care of reacting on the next tick anyway.

I'll write a PR to illustrate what I mean

**Comment by bsouthga** [comment]
I'm not sure about the original intent with the `resolve` pattern, but promises actually don't resolve on the next tick -- they resolve on the next microtask: https://stackoverflow.com/questions/25915634/difference-between-microtask-and-macrotask-within-an-event-loop-context.

You can see this clearly by running the following code in node or the browser: 

```javascript
new Promise(res => { console.log(1); res(); }); 
setTimeout(() => console.log(2), 0); 
new Promise(res => { console.log(3); res(); }); 
console.log(4);
```

Which should output

```
1
3
4
2
```

**Comment by benjamingr** [comment]
I actually started with that (promisifying setTimeout and using a setTimeout(fn, 0)) but then I realized it was only deferred so things like setting options would work (the fetch must be dispatched after all the configuration is done - meaning waiting a microtick is plenty.

That said, neither Node.js nor the browsers actually makes any guarantees about the order of things, and it's possible for promise callbacks to run before _or after_ timers. Node.js explicitly makes no such guarantee about scheduling - only best attempt.
